tests/unit/test_parameters.py

```diff
@@ -9,15 +9,68 @@
 from databricks.sql.thrift_api.TCLIService.ttypes import (
     TSparkParameter,
     TSparkParameterValue,
+    TSessionHandle,
+    TOpenSessionResp,
 )
 from databricks.sql.utils import DbSqlParameter, DbSqlType
 import pytest
 
-from decimal import Decimal
+from databricks.sql.thrift_backend import ThriftBackend
+from databricks.sql.thrift_api.TCLIService import ttypes
 
+from decimal import Decimal
+from databricks.sql.client import Connection
 from typing import List
 
 
+class TestSessionHandleChecks(object):
+    def test_get_session_handle(self):
+        assert (
+            Connection.get_protocol_version(
+                TOpenSessionResp(
+                    serverProtocolVersion=ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V7,
+                    sessionHandle=TSessionHandle(1, None),
+                )
+            )
+            == ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V7
+        )
+        assert (
+            Connection.get_protocol_version(
+                TOpenSessionResp(
+                    serverProtocolVersion=ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V7,
+                    sessionHandle=TSessionHandle(
+                        1, ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V8
+                    ),
+                )
+            )
+            == ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V8
+        )
+        assert (
+            Connection.get_protocol_version(
+                TOpenSessionResp(
+                    serverProtocolVersion=ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V8,
+                    sessionHandle=TSessionHandle(1, None),
+                )
+            )
+            == ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V8
```

Comment:
This test assertion is effectively a duplicate of the first assertion currently on line 35. Consider removing or separating into a separate test with a justification for why this test is required in addition to the first one.